datacube/utils/geometry/tools.py

```diff
@@ -621,7 +635,9 @@ def compute_roi(src, dst, tr, pts_per_side, padding, align):
         is_st = is_affine_st(tr.linear)
 
         if tight_ok and is_st:
-            roi_src, roi_dst = box_overlap(src.shape, dst.shape, tr.back.linear)
+            # Calculate tolerance in source space scaled by resolution
+            tolerance = tuple(abs(tol*res) for res in src.resolution)
+            roi_src, roi_dst = box_overlap(src.shape, dst.shape, tr.back.linear, tolerance)
```

Comment:
`tr.back.linear` is already using pixel units, so I don't think multiplying by `src.resolution` is doing the right thing here.
I think you should just pass in single `tol` number to `box_overlap`.

Reply:
Yep you're right.